### PR TITLE
Admin panel

### LINKED ISSUE #15: Admin panel (without content)
Links to design:
- https://www.figma.com/file/4tRZhRlLQ9U8kwTSLVEqHD/Voirsy?node-id=540%3A27502
- https://www.figma.com/file/4tRZhRlLQ9U8kwTSLVEqHD/Voirsy?node-id=540%3A28862
- https://www.figma.com/file/4tRZhRlLQ9U8kwTSLVEqHD/Voirsy?node-id=540%3A28347
- https://www.figma.com/file/4tRZhRlLQ9U8kwTSLVEqHD/Voirsy?node-id=540%3A28388

### PR DESCRIPTION
**Pull request by zawolanski**
Closed #15 

Creating the admin panel which currently contains only JSON data fetched from local json-server instead of specific content.

If anywhere in an app you will need a redirect to the admin panel you should redirect the user to the `/salons` route. The app will check that the user has any salons and automatically redirects to the first of them.

### CODE REVIEW

File: src/components/AdminPanelNavigation/adminPanelNavigation.styled.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    border: 0,
+  },
+}));
+
+export const CustomCard = styled(Card)(() => ({
+  background: theme.palette.primary.main,
+
+  '& a': {
+    textDecoration: 'none',
+    color: theme.palette.common.white,
```
Comment (pawelsobkowski) [inline src/components/AdminPanelNavigation/adminPanelNavigation.styled.ts]: We have defined color for text in `theme.palette.primary.contrastText`

File: src/components/AdminPanelNavigation/adminPanelNavigation.styled.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  textTransform: 'uppercase',
+
+  '& a': {
+    textDecoration: 'none',
+    color: theme.palette.grey[700],
+  },
+
+  '& svg': {
+    width: theme.spacing(4.5),
+    height: theme.spacing(4.5),
```
Comment (pawelsobkowski) [inline src/components/AdminPanelNavigation/adminPanelNavigation.styled.ts]: I don't think that using spacing to set those values is a good idea.
> Use the theme.spacing() helper to create consistent spacing between the elements of your UI.

File: src/components/AdminPanelNavigation/index.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  useEffect(() => handleClose(), [location]);
+
+  return (
+    <CustomDrawer open={isMenuOpen} variant="temporary">
+      <CustomToolbar>
+        <IconButton onClick={handleClose}>
+          <CloseOutlined />
+        </IconButton>
+      </CustomToolbar>
+      <Box p={2}>
```
Comment (pawelsobkowski) [inline src/components/AdminPanelNavigation/index.tsx]: If you want to use that mystical acronym, please leave at least a comment on what it means 🙂

File: src/components/Header/header.styles.ts [hunk 4]
```diff
@@ -16,3 +19,22 @@ export const CustomToolbar = styled(Toolbar)(() => ({
 export const CustomButton = styled(Button)(() => ({
   color: '#fff',
 }));
+
+export const CustomMenuButton = styled(IconButton)(() => ({
+  paddingLeft: 0,
+  paddingRight: 20,
```
Comment (pawelsobkowski) [inline src/components/Header/header.styles.ts]: Unit? Spacing?

File: src/templates/Salons/salons.navtabs.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  const routeMatch = useRouteMatch([
+    '/salons/:salonId/schedule',
+    '/salons/:salonId/portfolio',
+    '/salons/:salonId/edit',
+  ]);
+  const currentTab = routeMatch?.path;
+  const [translation] = useTranslation();
+
+  return (
+    <Box pl={matches ? 0 : 2}>
```
Comment (pawelsobkowski) [inline src/templates/Salons/salons.navtabs.tsx]: I don't know what it is 🤷‍♂️

File: src/templates/Salons/salons.styled.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  [theme.breakpoints.up('sm')]: {
+    height: `calc(100vh - ${(theme.mixins.toolbar['@media (min-width:600px)'] as any).minHeight}px)`,
+  },
+
+  '@media (orientation: landscape)': {
+    height: `calc(100vh - ${
+      (theme.mixins.toolbar['@media (min-width:0px) and (orientation: landscape)'] as any).minHeight
+    }px)`,
+  },
+}));
```
Comment (pawelsobkowski) [inline src/templates/Salons/salons.styled.ts]: I don't see that those are used in the component.

File: src/templates/Salons/index.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  if (isFetching) return <Spinner />;
+
+  if (pathname === '/salons') return <Redirect to={`/salons/${data[0]._id || 1}/edit`} />;
+
+  return (
+    <Stack direction="row">
+      <CustomSalonsNavigation p={2}>
+        <SalonsNavigation data={data} />
+      </CustomSalonsNavigation>
+      <Box width="100%" display="flex" flexDirection="column" minHeight="400px">
```
Comment (pawelsobkowski) [inline src/templates/Salons/index.tsx]: It is not required to use width or height everywhere. In that case, `minHeight` is needless and `width` can be replaced by `flexGrow`. Thanks to that, the application will be more responsive in the default way.

**Review by pawelsobkowski — CHANGES_REQUESTED** [review]

File: src/templates/Salons/salons.styled.ts (hunk 6 above)
Comment (zawolanski) [reply src/templates/Salons/salons.styled.ts]: Yes, I move it somewhere else and forgot to remove it.

**Review by pawelsobkowski — APPROVED** [review]

**Pull request merged**